dags/catalyst.py

```diff
@@ -0,0 +1,59 @@
+"""
+DAG to schedule generation of performance reports for recently completed nimbus experiments.
+
+See the [catalyst repository](https://github.com/mozilla/catalyst).
+
+*Triage notes*
+
+This app will perform some bigquery queries, and generate statistical reports based on that data which are
+then published to https://protosaur.dev/perf-reports/index.html.
+
+Occasionally, the app can fail if it cannot understand the nimbus experiment configuration properly, but most
+of these cases should have coverage by now.  Other issues that may occur is if the sample size is too low which
+may cause some problems during the statistical analysis.  I'm still looking for more examples of these situations
+to fix.
+
+Generally, there should be minimal triage necessary for failures unless it's related to infrastructure issues.
+Any failures related to the app execution itself will be taken care of directly by the performance team.
+
+"""
+
+from datetime import datetime, timedelta
+
+from airflow import DAG
+
+from operators.gcp_container_operator import GKEPodOperator
+from utils.tags import Tag
+
+default_args = {
+    "owner": "dpalmeiro@mozilla.com",
+    "email": [
+        "dpalmeiro@mozilla.com",
+    ],
+    "depends_on_past": False,
+    "start_date": datetime(2025, 3, 18),
+    "email_on_failure": True,
+    "email_on_retry": True,
+    "retries": 1,
+    "retry_delay": timedelta(minutes=30),
+}
+
+tags = [Tag.ImpactTier.tier_2]
+
+with DAG(
+    "catalyst",
+    default_args=default_args,
+    schedule_interval="0 4 * * *",
+    doc_md=__doc__,
+    tags=tags,
+) as dag:
+    # Built from repo https://github.com/mozilla/catalyst
+    catalyst_image = "gcr.io/moz-fx-data-experiments/catalyst:latest"
+
+    catalyst_run = GKEPodOperator(
```

Comment:
What do you think about having this dependent on an upstream DAG or task? This would ensure that new data is available before this task runs. I'm guessing you could do the same upstream tasks as Jetstream since I expect that it needs the same (or similar) data.

Reply:
See https://github.com/mozilla/telemetry-airflow/blob/main/dags/jetstream.py#L86-L145 for how the jetstream DAG configures this.

Reply:
Hmm.  What would the value of doing this be?  The project currently checks the nimbus list for completed experiments and then starts working on any recently completed ones that fit the requirements.  Do you propose separating that step out and doing it independently as a separate DAG?

Reply:
I see.  If there is a slightly delay for telemetry data to come in, then that is a valid concern.  For now, I can adjust the experiment search so that it only kicks off an analysis for experiments that have finished a day or two prior.  I do agree it might be useful to add a separate upstream DAG to do this, but I'm a bit eager to mostly get some kind of performance coverage at the moment since we don't have any right now.  What do you think?

Reply:
I think it would be pretty safe to wait on the same telemetry that Jetstream waits on, which isn't necessarily an exact overlap but includes general events and client info. If you'd prefer to delay by a day that's alright too, but I think this should be a pretty easy thing to put in here now without having to create additional DAGs or anything. You can just copy those lines I linked above and swap out the jetstream variable name in the last part (`jetstream_run.set_upstream(` --> `catalyst_run.set_upstream(`), so I think this would be even quicker than updating the code logic.